datalab/stackdriver/monitoring/_metric.py

```diff
@@ -0,0 +1,102 @@
+
+  _DISPLAY_HEADERS = ('Metric type', 'Display name', 'Kind', 'Value', 'Unit',
+                      'Labels')
+
+  def __init__(self, filter_string=None, type_prefix=None,
+               project_id=None, context=None):
+    """Initializes the MetricDescriptors based on the specified filters.
+
+    Args:
+      filter_string: An optional filter expression describing the resource
+        descriptors to be returned.
+      type_prefix: An optional prefix constraining the selected metric types. This
+        adds ``metric.type = starts_with("<prefix>")`` to the filter.
+      project_id: An optional project ID or number to override the one provided
+          by the context.
+      context: An optional Context object to use instead of the global default.
+    """
+    self._client = _utils.make_client(project_id, context)
+    self._filter_string = filter_string
+    self._type_prefix = type_prefix
+    self._descriptors = None
+
+  def list(self, pattern='*'):
+    """Returns a list of metric descriptors that match the filters.
+
+    Args:
+      pattern: An optional pattern to further filter the descriptors. This can
+        include Unix shell-style wildcards. E.g. "compute*", "*cpu/load_??m".
+
+    Returns:
+      A list of MetricDescriptor objects that match the filters.
+    """
+    if self._descriptors is None:
+      self._descriptors = self._client.list_metric_descriptors(
+          filter_string=self._filter_string, type_prefix=self._type_prefix)
+    return [metric for metric in self._descriptors
+            if fnmatch.fnmatch(metric.type, pattern)]
+
+  def as_dataframe(self, pattern='*', max_rows=None):
+    """Creates a pandas dataframe from the descriptors that match the filters.
+
+    Args:
+      pattern: An optional pattern to further filter the descriptors. This can
+        include Unix shell-style wildcards. E.g. "compute*", "*/cpu/load_??m".
+      max_rows: The maximum number of descriptors to return. If None, return all.
+
+    Returns:
+      A pandas dataframe containing matching metric descriptors.
+    """
+    data = []
+    for i, metric in enumerate(self.list(pattern)):
+      if max_rows is not None and i >= max_rows:
+        break
+      labels = ', '. join([l.key for l in metric.labels])
+      data.append([
+          metric.type, metric.display_name, metric.metric_kind,
+          metric.value_type, metric.unit, labels])
+
+    return pandas.DataFrame(data, columns=self._DISPLAY_HEADERS)
+
+  def table(self, pattern='*', max_rows=None):
+    """Visualize the matching descriptors as an HTML table.
+
+    Args:
+      pattern: An optional pattern to further filter the descriptors. This can
+        include Unix shell-style wildcards. E.g. "aws*", "*cluster*".
+      max_rows: The maximum number of descriptors to display. If None, display
+        all.
+
+    Returns:
+      The HTML rendering for a table of matching metric descriptors.
+    """
+    from . import _visualization
```

Comment:
Yes - sorry about that.

This is a lingering import from trying to not import IPython unless required.